### ISSUE TITLE
Test coverage of other packages should not be reported

### ISSUE THREAD

**Issue by linzhp** [bazelbuild/rules_go#2636]
When `bar` imports `cover`, the test coverage of `cover` reported by `go test -cover` does not include the part covered by tests in `bar`:

```bash
$ git clone git@github.com:linzhp/bazel_examples.git
$ cd bazel_examples
$ git checkout dcb322f
$ go test -cover github.com/linzhp/bazel_examples/cover github.com/linzhp/bazel_examples/cover/bar
ok  	github.com/linzhp/bazel_examples/cover	1.637s	coverage: 50.0% of statements
ok  	github.com/linzhp/bazel_examples/cover/bar	2.610s	coverage: 0.0% of statements
```

However, the coverage report of `bar` generated by Bazel/rules_go includes the coverage of `cover` when both test targets are running at the same time:

```bash
$ bazel coverage //cover/...
INFO: Using default value for --instrumentation_filter: "//cover[/:]".
INFO: Override the above default with --instrumentation_filter
INFO: Build options --collect_code_coverage, --instrumentation_filter, and --test_timeout have changed, discarding analysis cache.
INFO: Analyzed 4 targets (53 packages loaded, 7113 targets configured).
INFO: Found 2 targets and 2 test targets...
INFO: Elapsed time: 11.679s, Critical Path: 3.63s
INFO: 15 processes: 15 darwin-sandbox.
INFO: Build completed successfully, 27 total actions
//cover:go_default_test                                                  PASSED in 1.7s
  /private/var/tmp/_bazel_zplin/1b23f78923ad6c0028e8aab1a0a6d5fb/execroot/__main__/bazel-out/darwin-fastbuild/testlogs/cover/go_default_test/coverage.dat
//cover/bar:go_default_test                                              PASSED in 2.1s
  /private/var/tmp/_bazel_zplin/1b23f78923ad6c0028e8aab1a0a6d5fb/execroot/__main__/bazel-out/darwin-fastbuild/testlogs/cover/bar/go_default_test/coverage.dat

Executed 2 out of 2 tests: 2 tests pass.
INFO: Build completed successfully, 27 total actions
$ cat bazel-testlogs/cover/bar/go_default_test/coverage.dat

mode: set
github.com/linzhp/bazel_examples/cover/bar/bar.go:5.12,7.2 1 0
github.com/linzhp/bazel_examples/cover/foo.go:5.13,7.2 1 0
github.com/linzhp/bazel_examples/cover/foo.go:9.13,11.2 1 1
```

If we want to see a combined report of test coverage from both packages, the test coverage of `cover` may become inconsistent with `go tool -cover`:

```bash
$ go get github.com/wadey/gocovmerge
$ gocovmerge bazel-testlogs/cover/bar/go_default_test/coverage.dat bazel-testlogs/cover/go_default_test/coverage.dat > /tmp/cover.out
$ go get github.com/axw/gocov/gocov
$ gocov convert /tmp/cover.out| gocov report

github.com/linzhp/bazel_examples/cover/foo.go	 Foo1	 100.00% (1/1)
github.com/linzhp/bazel_examples/cover/foo.go	 Foo2	 100.00% (1/1)
github.com/linzhp/bazel_examples/cover		 ----	 100.00% (2/2)

github.com/linzhp/bazel_examples/cover/bar/bar.go	 Bar	 0.00% (0/1)
github.com/linzhp/bazel_examples/cover/bar		 ---	 0.00% (0/1)

Total Coverage: 66.67% (2/3)
```

Arguably, we may clean up `bazel-testlogs/cover/bar/go_default_test/coverage.dat` and remove the lines related to other packages before merging the coverage reports. However, if those lines are not needed, should they be produced at all?

I am not sure if this is the same issue as #1467, as I don't see more packages instrumented than the default `--instrumentation_filter`.

### What version of rules_go are you using?

0.24.0

### What version of gazelle are you using?

0.22.0

### What version of Bazel are you using?

3.4.1

### Does this issue reproduce with the latest releases of all the above?

Yes

### What operating system and processor architecture are you using?

Darwin amd64

**Comment by jayconrod** [comment]
I think this is expected behavior for Bazel coverage? Bazel tells rules which targets should be instrumented for coverage, so we just follow what it says. In theory, coverage data is supposed to be merged from all tests, but I don't believe that's ever worked for Starlark rules. Until we're unblocked there, I'm not going to spend much time working on rules_go coverage.

In any case, I don't think it's a bug that Bazel coverage would instrument different packages than `go test -cover`. The two systems have different ideas of how coverage should work and what should be instrumented.

**Comment by linzhp** [comment]
OK. Then I will clean up `coverage.dat` to get the same behavior as `go test -cover`.
